File: SampleCode/RestApi/Clients/Java/src/main/java/io/swagger/client/model/LibraryItemLinkType.java
/*
 * Emby Server REST API
 * 
 */

package io.swagger.client.model;

import java.util.Objects;
import java.util.Arrays;
import com.google.gson.annotations.SerializedName;
import java.io.IOException;
import com.google.gson.TypeAdapter;
import com.google.gson.annotations.JsonAdapter;
import com.google.gson.stream.JsonReader;
import com.google.gson.stream.JsonWriter;

/**
 * Gets or Sets Library.ItemLinkType
 */
@JsonAdapter(LibraryItemLinkType.Adapter.class)
public enum LibraryItemLinkType {
  ARTISTS("Artists"),
  ALBUMARTISTS("AlbumArtists"),
  GENRES("Genres"),
  STUDIOS("Studios"),
  TAGS("Tags"),
  COMPOSERS("Composers"),
  COLLECTIONS("Collections"),
  ALBUMS("Albums"),
  COLLECTIONFOLDERS("CollectionFolders");

  private String value;

  LibraryItemLinkType(String value) {
    this.value = value;
  }

  public String getValue() {
    return value;
  }

  @Override
  public String toString() {
    return String.valueOf(value);
  }

  public static LibraryItemLinkType fromValue(String input) {
    for (LibraryItemLinkType b : LibraryItemLinkType.values()) {
      if (b.value.equals(input)) {
        return b;
      }
    }
    return null;
  }

  public static class Adapter extends TypeAdapter<LibraryItemLinkType> {
    @Override
    public void write(final JsonWriter jsonWriter, final LibraryItemLinkType enumeration) throws IOException {
      jsonWriter.value(String.valueOf(enumeration.getValue()));
    }

    @Override
    public LibraryItemLinkType read(final JsonReader jsonReader) throws IOException {
      Object value = jsonReader.nextString();
      return LibraryItemLinkType.fromValue((String)(value));
    }
  }
}

File: SampleCode/RestApi/Clients/Net.RestSharp/EmbyClient.Dotnet/Model/LibraryItemLinkType.cs
/*
 * EmbyClient.Dotnet
 */

using System;
using System.Linq;
using System.IO;
using System.Text;
using System.Text.RegularExpressions;
using System.Collections;
using System.Collections.Generic;
using System.Collections.ObjectModel;
using System.Runtime.Serialization;
using Newtonsoft.Json;
using Newtonsoft.Json.Converters;
using SwaggerDateConverter = EmbyClient.Dotnet.Client.SwaggerDateConverter;

namespace EmbyClient.Dotnet.Model
{
    /// <summary>
    /// Defines Library.ItemLinkType
    /// </summary>
    [JsonConverter(typeof(StringEnumConverter))]
        public enum LibraryItemLinkType
    {
        /// <summary>
        /// Enum Artists for value: Artists
        /// </summary>
        [EnumMember(Value = "Artists")]
        Artists = 1,
        /// <summary>
        /// Enum AlbumArtists for value: AlbumArtists
        /// </summary>
        [EnumMember(Value = "AlbumArtists")]
        AlbumArtists = 2,
        /// <summary>
        /// Enum Genres for value: Genres
        /// </summary>
        [EnumMember(Value = "Genres")]
        Genres = 3,
        /// <summary>
        /// Enum Studios for value: Studios
        /// </summary>
        [EnumMember(Value = "Studios")]
        Studios = 4,
        /// <summary>
        /// Enum Tags for value: Tags
        /// </summary>
        [EnumMember(Value = "Tags")]
        Tags = 5,
        /// <summary>
        /// Enum Composers for value: Composers
        /// </summary>
        [EnumMember(Value = "Composers")]
        Composers = 6,
        /// <summary>
        /// Enum Collections for value: Collections
        /// </summary>
        [EnumMember(Value = "Collections")]
        Collections = 7,
        /// <summary>
        /// Enum Albums for value: Albums
        /// </summary>
        [EnumMember(Value = "Albums")]
        Albums = 8,
        /// <summary>
        /// Enum CollectionFolders for value: CollectionFolders
        /// </summary>
        [EnumMember(Value = "CollectionFolders")]
        CollectionFolders = 9,
        /// <summary>
        /// Enum LiveTVSeries for value: LiveTVSeries
        /// </summary>
        [EnumMember(Value = "LiveTVSeries")]
        LiveTVSeries = 10,
        /// <summary>
        /// Enum GameSystems for value: GameSystems
        /// </summary>
        [EnumMember(Value = "GameSystems")]
        GameSystems = 11    }
}
